File: jupyter2/ps/recursion/continued_fraction/ml/continued_fraction.ml
(** begin my answer *)

(** begin hidden *)
let rec continued_fraction a n =
  if n = 0 then
    1.0
  else
    1.0 /. (a +. continued_fraction a (n - 1))
(** end hidden *)
(** end my answer *)

let main () =
  assert (Float.abs (continued_fraction 2.0 100 -. 0.4142136) < 1.0e-6);
  assert (Float.abs (continued_fraction 3.0 100 -. 0.3027756) < 1.0e-6);
  assert (Float.abs (continued_fraction 4.0 100 -. 0.2360680) < 1.0e-6);
  Printf.printf "OK\n"
;;

main()

File: jupyter2/ps/recursion/continued_fraction/jl/continued_fraction.jl
### begin my answer

### begin hidden
function continued_fraction(a, n)
    if n == 0
        1.0
    else
        1.0 / (a + continued_fraction(a, n - 1))
    end
end
### end hidden
### end my answer

function main()
    @assert abs(continued_fraction(2.0, 100) - 0.4142136) < 1.0e-6
    @assert abs(continued_fraction(3.0, 100) - 0.3027756) < 1.0e-6
    @assert abs(continued_fraction(4.0, 100) - 0.2360680) < 1.0e-6
    println("OK")
end

main()
